### ISSUE TITLE
Outlook 2010: Different Output occasionally Given when Opening

### ISSUE THREAD

**Issue by elliott94** [nvaccess/nvda#10686]
### Steps to reproduce:

When opening Outlook 2010, I've noticed that on ocasion NVDA will sometimes display the message columns in a different order - example below:

Message Unread Importance High, From Firstname Lastname, Subject Subject of Message, Received 09:59, Size 7 KB, Flag Status Unflagged, 

This is instead of simply announcing "Unread from Name" along with the time that the message was received, along with the size of the message; from previous experience, information as to whether the message is flagged or not shouldn't be output. The only way that I've been able to restore standard reading is by restarting either NVDA or Outlook.

#### NVDA version:
2019.2.1

#### Windows version:

Windows 7/10

**Comment by Adriani90** [comment]
I cannot reproduce this. Does it also happen with addons disabled? Have you tried to run Outlook and Nvda as admin? And finally, do you have only one version of Outlook on your machine?

**Comment by DrSooom** [comment]
After I updated to NVDA 2019.2.1, I sometimes also have this behavior with Microsoft® Outlook® 2010 (14.0.7236.5000) SP2 MSO (14.0.7237.5000) on Windows 7 64-Bit, but it's very seldom.

@elliott94: Please list all your installed NVDA add-ons. This issue could be caused by the [Audio Themes 5.1 add-on](https://addons.nvda-project.org/addons/AudioThemes.en.html). But I haven't checked that out right now.

**Comment by DrSooom** [comment]
This issue could also be caused by using multiple PST files, which are located on different locations. In other words: If Outlook 2010 has to wait for the awakening of the HDD (was in sleep mode before), the startup process of Outlook 2010 will take more time. Just a thought. Deeper tests are pending.

**Comment by Adriani90** [comment]
@elliott94 are you sure that the order of columns is just reported differently by NVDA? Or is is also visually changed randomly?

**Comment by Adriani90** [comment]
To be honnest, I cannot reproduce this issue on Outllok 2010 or outlook 365.

**Comment by DrSooom** [comment]
@elliott94: Are you using IMAP or POP3 accounts?

In my case: Two POP3 accounts (one PST file on a SSD and one on a HDD due to a free upgrade from Microsoft Office 2007 to 2010 in June 2010). I have a separate e-mail account for my smartphones, which is set up there as IMAP, but I'm not using it in Microsoft Outlook 2010.

**Comment by DrSooom** [comment]
Before I forget: Ensure that the Outlook 2010 main window is always maximized (Win+ArrowUp).

**Comment by elliott94** [comment]
Apologies. Just to confirm - I don't have any add-ons present. All tests were performed with a single Exchange account and Cached Mode disabled - so that the Outlook client is forced to fetch new mail from the Exchange server each time - it doesn't rely on the presence of PST/OST files.

**Comment by elliott94** [comment]
Interestingly, the following errors are consistently generated in my log file whenever Outlook is first launched:

ERROR - RPC process 7396 (nvda_slave.exe) (09:20:23.062):
__main__.main:
slave error
Traceback (most recent call last):
  File "nvda_slave.pyw", line 94, in main
  File "comHelper.pyc", line 22, in _lresultFromGetActiveObject
  File "comtypes\client\__init__.pyc", line 173, in GetActiveObject
  File "comtypes\GUID.pyc", line 78, in from_progid
  File "_ctypes/callproc.c", line 950, in GetResult
WindowsError: [Error -2147221005] Invalid class string
ERROR - RPC process 5364 (nvda_slave.exe) (09:20:25.164):
__main__.main:
slave error
Traceback (most recent call last):
  File "nvda_slave.pyw", line 94, in main
  File "comHelper.pyc", line 22, in _lresultFromGetActiveObject
  File "comtypes\client\__init__.pyc", line 173, in GetActiveObject
  File "comtypes\GUID.pyc", line 78, in from_progid
  File "_ctypes/callproc.c", line 950, in GetResult
WindowsError: [Error -2147221005] Invalid class string
ERROR - appModules.outlook.AppModule._get_nativeOm (09:20:25.184):
Failed to get native object model
Traceback (most recent call last):
  File "appModules\outlook.pyc", line 117, in _get_nativeOm
  File "comHelper.pyc", line 58, in getActiveObject
RuntimeError: Helper process unable to get object; see log for details

**Comment by DrSooom** [comment]
See also: Issue #8961

**Comment by Adriani90** [comment]
in #8961 is a different error, they are not really related as far as I can understand.
@leonardder do you think this is yet another comtypes issue here? I finally could reproduce this issue ocasionally, more often in Outlook 2010 than in Outlook 365.

This is realy hard to investigate because it is not reproducible in a reliable way. I could only reproduce this at my work place but not at home. So It could also be a provilege issue here, on the work machine I can work only with a local user account, no admin rights and very limited permissions at all.

**Comment by Adriani90** [comment]
I could resolve this problem by re-creating the Outlook profile like in this tutorial:
https://www.howto-outlook.com/faq/newprofile.htm#create

Now the profile is clean and not only the columns are now everytime in correct order, but NVDA is working much faster now in the message list with the fresh profile.

Closing as works for me. @elliott94 please report if you still have the problem after re-creating your Ouglook profile and we can reopen. Thanks.
